packages/contracts/src/contracts/current/multisig/MultiSigWalletWithTimeLockExceptRemoveAuthorizedAddress/MultiSigWalletWithTimeLockExceptRemoveAuthorizedAddress.sol

```diff
@@ -35,21 +39,36 @@ contract MultiSigWalletWithTimeLockExceptRemoveAuthorizedAddress is MultiSigWall
     /// @param _owners List of initial owners.
     /// @param _required Number of required confirmations.
     /// @param _secondsTimeLocked Duration needed after a transaction is confirmed and before it becomes executable, in seconds.
-    /// @param _tokenTransferProxy Address of TokenTransferProxy contract.
+    /// @param _assetProxyContracts Array of AssetProxy contract addresses.
     function MultiSigWalletWithTimeLockExceptRemoveAuthorizedAddress(
-        address[] _owners,
-        uint _required,
-        uint _secondsTimeLocked,
-        address _tokenTransferProxy)
+        address[] memory _owners,
+        uint256 _required,
+        uint256 _secondsTimeLocked,
+        address[] memory _assetProxyContracts)
         public
         MultiSigWalletWithTimeLock(_owners, _required, _secondsTimeLocked)
     {
-        TOKEN_TRANSFER_PROXY_CONTRACT = _tokenTransferProxy;
+        for (uint256 i = 0; i < _assetProxyContracts.length; i++) {
+            require(_assetProxyContracts[i] != address(0));
+            isAssetProxyRegistered[_assetProxyContracts[i]] = true;
+        }
+    }
+
+    /// @dev Sets approval for calling removeAuthorizedAddress on an AssetProxy contract without a timelock.
```

Comment:
`Registers or deregisters an AssetProxy for future removeAuthorizedAddress without a timelock`